shortcircuit-api/src/main/java/com/powsybl/shortcircuit/ShortCircuitAnalysisResult.java

```diff
@@ -30,9 +30,16 @@
 
     private final List<LimitViolation> limitViolations = new ArrayList<>();
 
-    public ShortCircuitAnalysisResult(List<FaultResult> faultResults, List<LimitViolation> limitViolations) {
+    private List<ContributionResult> contributionResults;
+
+    public ShortCircuitAnalysisResult(List<FaultResult> faultResults, List<LimitViolation> limitViolations, List<ContributionResult> contributionResults) {
         this.faultResults.addAll(Objects.requireNonNull(faultResults));
         this.limitViolations.addAll(Objects.requireNonNull(limitViolations));
+        this.contributionResults = contributionResults;
```

Comment:
Same here: better to make an immutable copy, + we should check not null if we go for an empty list instead of null.

Actually we should do it also for the other existing fields.